### PR TITLE
feat: add remix style routing example

### PR DESCRIPTION
**Pull request by lxy-yz**
Following https://remix.run/docs/en/v1/api/conventions#route-file-conventions added remix style routes that supports:

- basic routes
- layout routes
- dynamic routes
- catch all routes
- pathless routes
- routes with escaping characters

### CODE REVIEW

File: src/resolvers/react.ts [hunk 1]
```diff
@@ -56,41 +57,29 @@ export async function resolveReactRoutes(ctx: PageContext) {
...
-      const isCatchAll = isCatchAllRoute(node, nuxtStyle)
-      const normalizedName = isDynamic
-        ? nuxtStyle
-          ? isCatchAll ? 'all' : node.replace(/^_/, '')
-          : node.replace(/^\[(\.{3})?/, '').replace(/\]$/, '')
-        : node
-      const normalizedPath = normalizeCase(normalizedName, caseSensitive)
 
       const route: Route = {
+        caseSensitive,
```
Comment (hannoeru) [inline src/resolvers/react.ts:67]: We should cleanup this in prepare route function, otherwise this will include in generated routes.

Comment (lxy-yz) [reply src/resolvers/react.ts:67]: `caseSensitive` is supported by react router v6 by default https://reactrouterdotcom.fly.dev/docs/en/v6/api#routes-and-route

```ts
interface RouteProps {
  caseSensitive?: boolean;
  children?: React.ReactNode;
  element?: React.ReactElement | null;
  index?: boolean;
  path?: string;
}
```

Comment (hannoeru) [reply src/resolvers/react.ts:67]: Oh... I didn't know that, haha

File: src/types.ts [hunk 2]
```diff
@@ -45,6 +45,12 @@ interface Options {
    * @default false
    */
   nuxtStyle: boolean
+  /**
+   * Use Remix style route naming
+   * https://remix.run/docs/en/v1/api/conventions#route-file-conventions
+   * @default false
+   */
+  remixStyle: boolean
```
[the remark this answers is not in the document]
Comment (lxy-yz) [reply src/types.ts:53]: Thanks for the suggestion, makes sense to me, will let the author to decide, also feel free to contribute as you might have more context about it.

[the remark this answers is not in the document]
Comment (hannoeru) [reply src/types.ts:53]: I personally prefer merge to one options.
@lxy-yz I think maybe we can do this in other PR, depends on you.

[the remark this answers is not in the document]
Comment (lxy-yz) [reply src/types.ts:53]: Sure i will merge the option given i am on this and fixing the CI error in the meantime.

**Comment by hannoeru** [comment]
@lxy-yz looks like lock file is broken, can you fix that?

**Comment by lxy-yz** [comment]
> @lxy-yz looks like lock file is broken, can you fix that?

I've tried to reproduce it locally but without luck. The error message in CI `ERR_PNPM_LOCKFILE_MISSING_DEPENDENCY  Broken lockfile: no entry for '/@vitejs/plugin-react/1.1.4' in pnpm-lock.yaml` does align with what I was seeing on local, suggestions?

**Comment by N0tExisting** [comment]
@lxy-yz as can be seen [here](https://github.com/hannoeru/vite-plugin-pages/runs/5194519202?check_suite_focus=true#step:5:13) you need to run a install.

It is also mentioned on the line above, that _This issue is probably caused by a badly resolved merge conflict._ (might have something to do with force-pushing?)

**Comment by lxy-yz** [comment]
> @lxy-yz as can be seen [here](https://github.com/hannoeru/vite-plugin-pages/runs/5194519202?check_suite_focus=true#step:5:13) you need to run a install.
> 
> It is also mentioned on the line above, that _This issue is probably caused by a badly resolved merge conflict._ (might have something to do with force-pushing?)

Thanks for taking a look. I've tried what the error message suggests. And lock file seems to look fine on local machine (with '/@vitejs/plugin-react/1.1.4' in pnpm-lock.yaml)

> This issue is probably caused by a badly resolved merge conflict. 

I didn't resolve any merge conflicts.

**Comment by hannoeru** [comment]
@lxy-yz I think run `pnpm update` can fix that.

**Pull request merged**